<launch>
    <!-- Step1:修改为自己的刚体名称，注意建立刚体时机头朝向！！ -->

    <arg name="rigid_name" default="uav1" />
    <arg name="Your_Topic_Name" default="/vision_odom" />

    <!-- Step2:启动vrpn_client_ros -->

    <include file="$(find vrpn_client_ros)/launch/sample.launch">
        <arg name="server" value="192.168.1.1"/>
    </include>

    <!-- Step3:启动 mavros -->

    <include file="$(find mavros)/launch/px4.launch">
        <arg name="fcu_url" value="/dev/ttyACM0:57600" />
        <!-- 这里可以选择通过WIFI远程连接地面站 -->
        <!-- （推荐）Option 1： 使用广播模式，局域网内的地面站都可以自动连接到-->
	<!--<arg name="gcs_url" value="udp-b://@"/>-->
	<arg name="gcs_url" value="udp://：@192.168.1.100:14551/>
        <!-- Option 2： 指定IP和端口号，假设地面站IP为192.168.1.100， UDP端口选择14550：-->
        <!-- <arg name="gcs_url" value="udp://@192.168.1.100:14550" /> -->
	    <arg name="tgt_system" value="3" />
    </include>

    <!-- Step4: 映射话题-->
    <node 
        name="relay_vrpn_node" 
        pkg="topic_tools" type="relay" 
        args="/vrpn_client_node/$(arg rigid_name)/pose /mavros/vision_pose/pose" 
        output="screen"
    />

    <!--（可选） Step5: 融合里程计。动捕没有里程计信息，手动融合，发布话题为/vision_odom-->
    <node   
        name="odom_publisher_node"
        pkg="odom_publisher_node" 
        type="odom_publisher_node"  
        output="screen" >
        <remap from="/vrpn_client_node/uav1/pose" to="/vrpn_client_node/$(arg rigid_name)/pose"/>
        <remap from="/vrpn_client_node/uav1/twist" to="/vrpn_client_node/$(arg rigid_name)/twist"/>
        <remap from="/vision_odom" to="$(arg Your_Topic_Name)"/>
    </node> 
    
    <!--（可选） Step6: 增加IMU和Odom的发布频率-->
    <node 
        name="mavcmd_imu" 
        pkg="mavros" 
        type="mavcmd" 
        args="long 511 105 5000 0 0 0 0 0 " 
        output="screen"
    />

    <node 
        name="mavcmd_odom" 
        pkg="mavros" 
        type="mavcmd" 
        args="long 511 31 5000 0 0 0 0 0 " 
        output="screen"
    />
</launch>
